src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java

```diff
@@ -89,6 +89,11 @@ public static int getNibble(byte b, int nibbleIndex)
                     "Nibble index can only be 0 or 1");
         }
     }
+    
```

Comment:
This method takes a byte array and produces a smaller byte array indicated by the arguments. If the operation of the method is not clear, please write a skeletal example using the same class. That should make the operation of this method very obvious.

Reply:
Please follow this advice of constructing minimum working example (MWE) for all the methods that are not clear.